source/common/http/http1/codec_impl.cc

```diff
@@ -1347,8 +1347,8 @@ Envoy::StatusOr<ParserStatus> ClientConnectionImpl::onHeadersCompleteBase() {
       }
     }
 
-    if (parser_->statusCode() == enumToInt(Http::Code::Continue)) {
-      pending_response_.value().decoder_->decode100ContinueHeaders(std::move(headers));
+    if (HeaderUtility::isSpecial1xx(*headers)) {
+      pending_response_.value().decoder_->decode1xxHeaders(std::move(headers));
```

Comment:
So any other 1xx header would currently just be handled as regular headers?

Reply:
yeah this is intentional for 101s but other codes get dropped in UpstreamRequest::decodeHeaders
honestly I think it'd be more clean to handle in the codecs, but then we'd be doing it in 3x places.